config.go

```diff
@@ -99,9 +99,29 @@ type YNAB struct {
 	// settings section
 	Token string `envconfig:"YNAB_TOKEN"`
 
+	// FromDate only import transactions from this date and onward. For
+	// example: 2006-01-02
+	FromDate Date `envconfig:"YNAB_FROM_DATE"`
+
 	// Set cleared status, possible values: cleared, uncleared, reconciled .
 	// Default is uncleared for historical reasons but recommend setting this
 	// to cleared because ynabber transactions are cleared by bank.
 	// They'd still be unapproved until approved in YNAB.
 	Cleared string `envconfig:"YNAB_CLEARED" default:"uncleared"`
+
+	ImportID ImportID
+}
+
+// ImportID can be either v1 or v2. All new users should use v2 because it
+// have a lower potability of making duplicate transactions. But v1 remains
+// the default to retain backwards compatibility.
+//
+// To migrate from v1 to v2 simply set the v2 to any date and all transactions
+// from and including that date will be using v2 of the import ID generator.
+type ImportID struct {
+	// V1 will be used from this date
+	V1 Date `envconfig:"YNAB_IMPORT_ID_V1" default:"1970-01-01"`
+
+	// V2 will be used from this date, for example: 2022-12-24
+	V2 Date `envconfig:"YNAB_IMPORT_ID_V2" default:"9999-01-01"`
```

Comment:
Could we have the date by default as something like 2023-03-30 or some other date in the fairly near future? We could assume current users will upgrade eventually ...

Reply:
I don't see how we could assume that. I want to gather learnings about this feature before making any upgrade assumptions or cutover dates.

Reply:
The best approach would be to actually use V2 imports by default for all new users and even for all new transactions. But I fail to see how this would be possible easily. The only way how this could be accomplished would be to query YNAB for existing transactions and see if there are any with an 'old' pattern. If none found use new, otherwise old ... but this becomes fairly complicated - I don't even know whether it is feasible at all. 

Perhaps, at some point, make these so that there are no defaults and the user must explicitly select this or the cutover date. We can even helpfully suggest if the parameter is missing on what to do. But this is another extra feature and will break current operation for existing users (although trivial to fix).